Week-3/App.js

```diff
@@ -0,0 +1,78 @@
+// ASSIGNMENT 3.1 ->
+function memoize(fn) {
+    const map = new Map();
+    return (...args) => {
+      var key = args.toString();
+      if (map.has(key)) return map.get(key);
+      if(args.length === 1) args.push(0);
+      map.set(key, fn(...args));
+      return map.get(key);
+    };
+}
+function add(a, b) {
+  console.log(a,b)
+  return a + b;
+}
+//Create a method called memoize such that:
+const memoizeAdd = memoize(add);
+//then calling.
+console.log(memoizeAdd(100, 100)); //returns 200
+console.log(memoizeAdd(100)); //returns 100
+console.log(memoizeAdd(100, 200)); //returns 300
```

Comment:
what if we change the position of params as in memorizeAdd(200,100) will it return from cache or compute again..it should return from cache.. please check